src/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -35,11 +35,24 @@ public class PushCommand : Command
 
             // Don't push symbols by default
             string source = ResolveSource(packagePath, ConfigurationDefaults.Instance.DefaultPushSource);
+            var pushEndpoint = await GetPushEndpointAsync(source);
 
-            var apiKey = GetApiKey(source);
-            if (String.IsNullOrEmpty(apiKey) && !IsFileSource(source))
+            if (string.IsNullOrEmpty(pushEndpoint))
             {
-                Console.WriteWarning(LocalizedResourceManager.GetString("NoApiKeyFound"), CommandLineUtility.GetSourceDisplayName(source));
+                var message = string.Format(
```

Comment:
Not really needed for a single arg, but add CultureInfo here to avoid build warnings.